File: src/main/java/me/onixdev/util/time/PlayerTimer.java
package me.onixdev.util.time;


import me.onixdev.user.OnixUser;

public class PlayerTimer {
    private OnixUser player;
    public int startTime;
    public int resetStreak;

    public PlayerTimer(OnixUser player) {
        this.player = player;
        this.reset();
    }

    public boolean wasReset() {
        return this.startTime == player.currentTick;
    }

    public boolean wasNotReset() {
        return this.startTime != player.currentTick;
    }

    public void reset() {
        if (getPassed() == 1) resetStreak++;
        else resetStreak = 0;
        this.startTime = player.currentTick;
    }

    public int getResetStreak() {
        return wasNotReset() ? 0 : resetStreak;
    }

    public long getPassed() {
        return player.currentTick - this.startTime;
    }

    public void add(int amount) {
        this.startTime -= amount;
    }

    public boolean hasPassed(long toPass) {
        return this.getPassed() >= toPass;
    }

    public boolean hasNotPassed(long toPass) {
        return this.getPassed() < toPass;
    }

    public boolean hasPassed(long toPass, boolean reset) {
        boolean passed = this.getPassed() >= toPass;
        if (passed && reset) reset();
        return passed;
    }

    public static boolean hasPassed(long startTime, long toPass) {
        return (System.currentTimeMillis() - startTime) >= toPass;
    }
}

File: src/main/java/me/onixdev/util/time/PlayerTimer.kt
package me.onixdev.util.time

import me.onixdev.user.OnixUser


class PlayerTimer(private val player: OnixUser) {
    var startTime: Int = 0
    var resetStreak1: Int = 0

    init {
        this.reset()
    }

    fun wasReset(): Boolean {
        return this.startTime == player.currentTick
    }

    fun wasNotReset(): Boolean {
        return this.startTime != player.currentTick
    }

    fun reset() {
        if (this.passed == 1L) resetStreak1++
        else resetStreak1 = 0
        this.startTime = player.currentTick
    }

    fun getResetStreak(): Int {
        return if (wasNotReset()) 0 else resetStreak1
    }

    val passed: Long
        get() = (player.currentTick - this.startTime).toLong()

    fun add(amount: Int) {
        this.startTime -= amount
    }

    fun hasPassed(toPass: Long): Boolean {
        return this.passed >= toPass
    }

    fun hasNotPassed(toPass: Long): Boolean {
        return this.passed < toPass
    }

    fun hasPassed(toPass: Long, reset: Boolean): Boolean {
        val passed = this.passed >= toPass
        if (passed && reset) reset()
        return passed
    }

}